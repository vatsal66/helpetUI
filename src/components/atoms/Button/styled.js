import styled from 'styled-components';

import { spreadSizes } from '@helpers/styles';

export const StyledButton = styled.button`
  cursor: pointer;
  padding: ${({ padding }) => (padding ? spreadSizes(padding) : '6px 12px')};
  border-radius: ${({ round }) => (round ? 25 : 0)}px;
  background-color: ${({ theme, backgroundColor }) =>
    backgroundColor ?? theme.colors.primary};
  color: ${({ theme, color }) => color ?? theme.colors.white};
  border: 0;
  outline: 0;

  &:disabled {
    background-color: ${({ theme }) => theme.colors.disabled};
    cursor: not-allowed;
  }
`;
