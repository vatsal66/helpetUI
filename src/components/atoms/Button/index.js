import React from 'react';
import PropTypes from 'prop-types';

import { StyledButton } from './styled';

const Button = ({
  className,
  children,
  padding,
  color,
  backgroundColor,
  upperCase,
  round,
  onClick,
  disabled,
  as,
}) => (
  <StyledButton
    {...{
      className,
      color,
      backgroundColor,
      padding,
      round,
      disabled,
      onClick,
      as,
    }}
  >
    {upperCase ? children.toUpperCase() : children}
  </StyledButton>
);

Button.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
  color: PropTypes.string,
  backgroundColor: PropTypes.string,
  padding: PropTypes.arrayOf(
    PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  ),
  upperCase: PropTypes.bool,
  round: PropTypes.bool,
  disabled: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
};

Button.defaultProps = {
  className: null,
  color: null,
  backgroundColor: null,
  upperCase: false,
  round: true,
  disabled: false,
  padding: null,
};

export default Button;
