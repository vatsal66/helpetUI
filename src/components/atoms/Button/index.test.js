import React from 'react';
import { fireEvent } from '@testing-library/react';

import renderWithTheme from '@mocks/renderWithTheme';

import Button from './index';

describe('<Button />', () => {
  const props = {
    children: 'Teste',
    onClick: jest.fn(),
  };

  it('should match snapshot file', () => {
    const { asFragment } = renderWithTheme(<Button {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should fired event on click the button', () => {
    const { getByText } = renderWithTheme(<Button {...props} />);
    fireEvent(
      getByText(props.children),
      new MouseEvent('click', { bubbles: true, cancelable: true }),
    );
  });
});
