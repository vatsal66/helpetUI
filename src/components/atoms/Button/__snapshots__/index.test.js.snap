// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot file 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm gpyKgC"
  >
    Teste
  </button>
</DocumentFragment>
`;
